components/ai_chat/core/browser/conversation_driver.cc

```diff
@@ -1052,9 +1062,94 @@ void ConversationDriver::PerformAssistantGeneration(
   auto data_completed_callback =
       base::BindOnce(&ConversationDriver::OnEngineCompletionComplete,
                      weak_ptr_factory_.GetWeakPtr(), current_navigation_id);
-  engine_->GenerateAssistantResponse(is_video, page_content, chat_history_,
-                                     input, std::move(data_received_callback),
-                                     std::move(data_completed_callback));
+  bool should_refine_page_content =
+      features::IsPageContentRefineEnabled() &&
+      page_content.length() > GetCurrentModel()
+                                  .options->get_leo_model_options()
```

Comment:
Doesn't this cause a crash for non-Leo models (aka custom models / BYOM)?

Reply:
ah yes, we need
```
     GetCurrentModel().options->is_custom_model_options()
          ? kCustomModelMaxPageContentLength
          : model.options->get_leo_model_options()->max_page_content_length;
```

Reply:
https://github.com/brave/brave-browser/issues/40686